### PR TITLE
[cli] [tabletserver/throttler] Migrate throttler flags to pflags

### LINKED ISSUE #10985: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/tabletserver/throttle`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/tabletserver/throttle/*.go
go/vt/vttablet/tabletserver/throttle/throttler.go:      throttleThreshold         = flag.Duration("throttle_threshold", 1*time.Second, "Replication lag threshold for default lag throttling")
go/vt/vttablet/tabletserver/throttle/throttler.go:      throttleTabletTypes       = flag.String("throttle_tablet_types", "replica", "Comma separated VTTablet types to be considered by the throttler. default: 'replica'. example: 'replica,rdonly'. 'replica' aways implicitly included")
go/vt/vttablet/tabletserver/throttle/throttler.go:      throttleMetricQuery       = flag.String("throttle_metrics_query", "", "Override default heartbeat/lag metric. Use either `SELECT` (must return single row, single value) or `SHOW GLOBAL ... LIKE ...` queries. Set -throttle_metrics_threshold respectively.")
go/vt/vttablet/tabletserver/throttle/throttler.go:      throttleMetricThreshold   = flag.Float64("throttle_metrics_threshold", math.MaxFloat64, "Override default throttle threshold, respective to -throttle_metrics_query")
go/vt/vttablet/tabletserver/throttle/throttler.go:      throttlerCheckAsCheckSelf = flag.Bool("throttle_check_as_check_self", false, "Should throttler/check return a throttler/check-self result (changes throttler behavior for writes)")
```

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

Updates throttler related flag references specified in #10985

## Related Issue(s)

Fixes #10985

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by rsajwani** [comment]
LGTM.. 
general comment: I would be good to add result of following command in your description. This helps in verify init() function.

go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ " <full import path of your package here> " {print $1}'

File: go/vt/vttablet/tabletserver/throttle/throttler.go [hunk 1]
```diff
@@ -22,6 +21,10 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
```
Comment (ajm188) [inline go/vt/vttablet/tabletserver/throttle/throttler.go:27]: nit: this blank line is unnecessary

**Review by ajm188 — APPROVED** [review]

**Comment by ajm188** [comment]
@rsajwani if you follow how/where the flags are actually used in the code, you'll see that only `NewThrottler` calls it, which is only imported by `vttablet` (and indirectly by `vtcombo`)

File: go/vt/vttablet/tabletserver/throttle/throttler.go [hunk 2]
```diff
@@ -62,12 +65,28 @@ const (
...
+)
 
+func init() {
+	servenv.OnParseFor("vtcombo", registerThrottlerFlags)
+	servenv.OnParseFor("vttablet", registerThrottlerFlags)
+}
+
+func registerThrottlerFlags(fs *pflag.FlagSet) {
+	fs.DurationVar(&throttleThreshold, "throttle_threshold", throttleThreshold, "Replication lag threshold for default lag throttling")
+	fs.StringVar(&throttleTabletTypes, "throttle_tablet_types", throttleTabletTypes, "Comma separated VTTablet types to be considered by the throttler. default: 'replica'. example: 'replica,rdonly'. 'replica' aways implicitly included")
```
Comment (ajm188) [inline go/vt/vttablet/tabletserver/throttle/throttler.go:83]: you should see if you can use `fs.StringSliceVar` here and remove the splitting below

Comment (rohit-nayak-ps) [reply go/vt/vttablet/tabletserver/throttle/throttler.go:83]: This list supports multiple delimiters: comma, semi-colon or space. 
```
// SplitDelimitedList splits a given string by comma, semi-colon or space, and returns non-empty strings
func SplitDelimitedList(s string) (list []string) {
```

File: go/vt/vttablet/tabletserver/throttle/throttler.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/vt/vttablet/tabletserver/throttle/throttler.go]: fixed, I had thought gofmt used to reorganize the imports canonically ...

Comment (ajm188) [reply go/vt/vttablet/tabletserver/throttle/throttler.go]: there are a wide variety of, uh, "arrangements" that `gofmt` accepts, so there isn't technically a "canonically".

for example, each of these are valid:

```go
import (
	"context"
	"flag"
	"fmt"
	"io"
	"sync"
	"time"

	"google.golang.org/grpc"

	"vitess.io/vitess/go/netutil"
	"vitess.io/vitess/go/vt/callerid"
	"vitess.io/vitess/go/vt/grpcclient"
	"vitess.io/vitess/go/vt/hook"
	"vitess.io/vitess/go/vt/log"
	"vitess.io/vitess/go/vt/logutil"
	"vitess.io/vitess/go/vt/mysqlctl/tmutils"
	"vitess.io/vitess/go/vt/topo/topoproto"
	"vitess.io/vitess/go/vt/vttablet/tmclient"
)
```

```go
import (
	"context"
	"io"
	"sync"

	"vitess.io/vitess/go/vt/log"

	"flag"

	"vitess.io/vitess/go/vt/callerid"

	"fmt"

	"google.golang.org/grpc"

	"vitess.io/vitess/go/netutil"
	"vitess.io/vitess/go/vt/grpcclient"
	"vitess.io/vitess/go/vt/hook"

	"time"

	"vitess.io/vitess/go/vt/logutil"
	"vitess.io/vitess/go/vt/mysqlctl/tmutils"
	"vitess.io/vitess/go/vt/topo/topoproto"
	"vitess.io/vitess/go/vt/vttablet/tmclient"
)
```

```go
import (
	"context"

	"github.com/spf13/pflag"

	"vitess.io/vitess/go/vt/mysqlctl/tmutils"

	"io"

	"vitess.io/vitess/go/vt/topo/topoproto"

	"sync"

	"vitess.io/vitess/go/vt/log"

	"flag"

	"vitess.io/vitess/go/vt/callerid"

	"fmt"

	"google.golang.org/grpc"

	"vitess.io/vitess/go/netutil"
	"vitess.io/vitess/go/vt/grpcclient"
	"vitess.io/vitess/go/vt/hook"

	"time"

	"vitess.io/vitess/go/vt/logutil"
	"vitess.io/vitess/go/vt/vttablet/tmclient"
)
```

now, obviously, this is at least partly aesthetic preference, and i'm being at least a little ridiculous with the examples i came up, but nonetheless i hope you agree the first one is the most pleasing / easily-scannable

**Pull request merged**